pkg/lib/workflow/workflowconfig/intent_signup_flow.go

```diff
@@ -0,0 +1,103 @@
+package workflowconfig
+
+import (
+	"context"
+
+	"github.com/iawaknahc/jsonschema/pkg/jsonpointer"
+
+	"github.com/authgear/authgear-server/pkg/lib/config"
+	"github.com/authgear/authgear-server/pkg/lib/workflow"
+	"github.com/authgear/authgear-server/pkg/util/uuid"
+	"github.com/authgear/authgear-server/pkg/util/validation"
+)
+
+func init() {
+	workflow.RegisterPublicIntent(&IntentSignupFlow{})
+}
+
+var IntentSignupSchema = validation.NewSimpleSchema(`
+{
+	"type": "object",
+	"additionalProperties": false,
+	"required": ["signup_flow"],
+	"properties": {
+		"signup_flow": { "type": "string" }
+	}
+}
+`)
+
+type IntentSignupFlow struct {
+	SignupFlow  string        `json:"signup_flow,omitempty"`
+	JSONPointer jsonpointer.T `json:"json_pointer,omitempty"`
+}
+
+func (*IntentSignupFlow) Kind() string {
+	return "workflowconfig.IntentSignupFlow"
+}
+
+func (*IntentSignupFlow) JSONSchema() *validation.SimpleSchema {
+	return IntentSignupSchema
+}
+
+func (i *IntentSignupFlow) CanReactTo(ctx context.Context, deps *workflow.Dependencies, w *workflow.Workflow) ([]workflow.Input, error) {
+	f, err := FindSignupFlow(deps.Config.Workflow, i.SignupFlow)
+	if err != nil {
+		return nil, err
+	}
+
+	// The list of nodes looks like
+	// 1 NodeDoCreateUser
+	// N Nodes for each N steps
+	// 1 IntentCreateSession
+	// So at the end of the flow, it will have 2+N nodes.
+	if len(w.Nodes) >= 2+len(f.Steps) {
+		return nil, workflow.ErrEOF
+	}
```

Comment:
It looks like this kind of assumptions can be easily broken, What if we check if the last node is `NodeDoCreateSession`?

Reply:
OK. But checking the type of the last node could also be fragile, for example, if we later on refactor to use IntentCreateSession instead of NodeDoCreateSession. The last node would a subworkflow with the intent IntentCreateSession.